reputation/reputation.go

```diff
@@ -0,0 +1,413 @@
+			row.name,
+			row.uptime,
+			row.auditSuccess,
+			row.auditFail,
+			row.latency,
+			row.amountOfDataStored,
+			row.falseClaims,
+			row.shardsModified,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	tx.Commit()
+
+	defer insertStmt.Close()
+}
+
+// side effect function that prints the rows from the query string
+func selectFromDB(db *sql.DB, selectString string) {
+	rows, err := db.Query(selectString)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for rows.Next() {
+		var name string
+		var timestamp string
+		var uptime int
+		var auditSuccess int
+		var auditFail int
+		var latency int
+		var amountOfDataStored int
+		var falseClaims int
+		var shardsModified int
+
+		err = rows.Scan(
+			&name,
+			&timestamp,
+			&uptime,
+			&auditSuccess,
+			&auditFail,
+			&latency,
+			&amountOfDataStored,
+			&falseClaims,
+			&shardsModified)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// side effect
+		fmt.Println(name,
+			timestamp,
+			uptime,
+			auditSuccess,
+			auditFail,
+			latency,
+			amountOfDataStored,
+			falseClaims,
+			shardsModified)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer rows.Close()
+}
+
+// function that returns a slice of reputation rows based on the query string
+func getRepRows(db *sql.DB, selectString string) []RepRow {
+	var res []RepRow
+
+	rows, err := db.Query(selectString)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for rows.Next() {
+		var name string
+		var timestamp string
+		var uptime int
+		var auditSuccess int
+		var auditFail int
+		var latency int
+		var amountOfDataStored int
+		var falseClaims int
+		var shardsModified int
+
+		err = rows.Scan(
+			&name,
+			&timestamp,
+			&uptime,
+			&auditSuccess,
+			&auditFail,
+			&latency,
+			&amountOfDataStored,
+			&falseClaims,
+			&shardsModified,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		currentRow := RepRow{
+			name,
+			timestamp,
+			uptime,
+			auditSuccess,
+			auditFail,
+			latency,
+			amountOfDataStored,
+			falseClaims,
+			shardsModified,
+		}
+
+		res = append(res, currentRow)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer rows.Close()
+
+	return res
+}
+
+// close sqlite3
+func cleanUpDB(db *sql.DB) {
+	defer db.Close()
+}
+
+// finds the ration of audit success from the success and failure fields of a given reputaion row struct
+func (row RepRow) auditSuccessRatio() float64 {
+	return float64(row.auditSuccess) / float64(row.auditSuccess+row.auditFail)
+}
+
+// naive formula for obtaining repuataion
+func (row RepRow) naiveRep() float64 {
+	var mutator int
+
+	if row.shardsModified > 0 {
+		mutator = 0
+	} else {
+		mutator = 1
+	}
+
+	return (float64(row.uptime*100) +
+		row.auditSuccessRatio() +
+		float64(row.latency) +
+		float64(row.amountOfDataStored) -
+		float64(row.falseClaims)) * float64(mutator)
+}
+
+// compares reputation rows and returns the greater reputation of the two
+func (row RepRow) greaterRep(other RepRow) RepRow {
```

Comment:
honestly, i'd do something more straightforward here like:

- first see if there are any false claims. nodes that have false claims are worse than nodes that don't.
- then compare audit ratios. whichever node has a better audit ratio is better
- if the audit ratios are the same, then look at uptime
- if everything else is equal, then look at latency, etc.